### PR TITLE
Add process flag preRegister &update zk-census doc

### PR DESCRIPTION
**Pull request by arnaucube**
- Add process flag preRegister
    - Update anonymous-voting/zk-census-proof doc replacing
    `CensusRegisterFunction` by `CensusOrigin`

[~The commit `Update missing zkCensusKey instead public-key for spec consistency` is from another PR (#55 ), this PR's branch is from that branch to avoid potential git conflicts~, Update: rebased to `master` as #55 is now merged]

### CODE REVIEW

**Review by NateWilliams2 — APPROVED** [review]
Made some minor edits, including adding `preRegister` to the gateway api example for processes. looks good to me.

File: src/architecture/protocol/anonymous-voting/zk-census-proof.md [hunk 1]
```diff
@@ -276,6 +280,12 @@ List of current types:
 
 
 # Annex
+#### Examples of flags combinations
+Below there are listed some common combinations of flags used when created a new process:
+- CSV with pre-register: `preRegister=true`, `CensusOrigin=OFF_CHAIN_TREE`
```
Comment (p4u) [inline src/architecture/protocol/anonymous-voting/zk-census-proof.md:285]: I'd not say here just CSV, its hard to understand. 

`Census Merkle Tree (from CSV file or private database) with pre-register:`

File: src/architecture/smart-contracts/process.md [hunk 2]
```diff
@@ -184,6 +185,10 @@ The process mode affects both the Vochain, the contract itself and even the meta
 
 It requires a prior process to share the encryption key with the users with the right to read the data. This will be likely be handled by the `User Registry`.
 
+##### PRE_REGISTER
+- `false` ⇒ The `CensusRoot` will be defined through a phase where the users need to register, using the `CensusRegisterProof` depending on the `CensusOrigin` of the process.
```
Comment (p4u) [inline src/architecture/smart-contracts/process.md:189]: false: There is not registration phase thus the census root used for validate votes is the censusRoot parameter defined by the newProcessTx transaction

true: There is a registration phase, the census root parameter defined by newProcessTx is used only during the registration (before the process startBlock is reached). The voting censusRoot will be built inside the Vochain when registerVoterKeyTx are send by the elegible voters

File: src/architecture/protocol/anonymous-voting/zk-census-proof.md (hunk 1 above)
Comment (arnaucube) [reply src/architecture/protocol/anonymous-voting/zk-census-proof.md]: :+1:  Added

File: src/architecture/smart-contracts/process.md (hunk 2 above)
Comment (arnaucube) [reply src/architecture/smart-contracts/process.md]: :+1:  Added

File: src/architecture/protocol/anonymous-voting/zk-census-proof.md [hunk 3]
```diff
@@ -149,7 +152,7 @@ sequenceDiagram
 opt Steps 1-2 for the csv processes
   Organizer->>Organizer: 0.1. create user login keys
   Organizer->>Organizer: 0.2. build merkleTree with login keys
-  Organizer->>Vochain: 1. create new voting process (newProcessTx) & define the CensusRegisterFunction=CSVCensus
+  Organizer->>Vochain: 1. create new voting process (newProcessTx) & define the CensusOrigin=CSVCensus
```
Comment (ed255) [inline src/architecture/protocol/anonymous-voting/zk-census-proof.md:155]: `CSVCensus` is not a valid CensusOrigin.  Maybe use `OFF_CHAIN_TREE` or `OFF_CHAIN_TREE_WEIGHTED`

File: src/architecture/protocol/anonymous-voting/zk-census-proof.md [hunk 4]
```diff
@@ -160,8 +163,8 @@ Note over Organizer,User: From here continues with the normal flow
 - 0.1. *[O]* Create user **login keys**
     - from csv data
 - 0.2. *[O]* Build **MerkleTree** with login keys
-- 1. *[O+V]* Create **new voting process** (newProcessTx) & define the CensusRegisterFunction=CSVCensus
-    - **CensusRegisterFunction** checks that:
+- 1. *[O+V]* Create **new voting process** (newProcessTx) & define the CensusOrigin=CSVCensus
```
Comment (ed255) [inline src/architecture/protocol/anonymous-voting/zk-census-proof.md:166]: Same as before

File: src/architecture/protocol/anonymous-voting/zk-census-proof.md [hunk 5]
```diff
@@ -160,8 +163,8 @@ Note over Organizer,User: From here continues with the normal flow
...
+    - **CensusOrigin** determines the checks:
```
Comment (ed255) [inline src/architecture/protocol/anonymous-voting/zk-census-proof.md:167]: The next line mentions `Ethereum Root`, but maybe you're refering to the `Census Root`?

File: src/architecture/protocol/anonymous-voting/zk-census-proof.md [hunk 6]
```diff
@@ -184,27 +188,27 @@ secret keys to the Vochain, avoiding to need the registration phase.
 
 sequenceDiagram
 opt Steps 1-2 for the Ethereum Storage Proofs processes
-  Organizer->>Vochain: 1.0. create new voting process (newProcessTx) & define the CensusRegisterFunction=EthStorageProofsCensus
+  Organizer->>Vochain: 1.0. create new voting process (newProcessTx) & define the CensusOrigin=EthStorageProofsCensus
```
Comment (ed255) [inline src/architecture/protocol/anonymous-voting/zk-census-proof.md:191]:
```suggestion
  Organizer->>Vochain: 1.0. create new voting process (newProcessTx) & define the CensusOrigin=ERC20
```

Following spec at https://github.com/vocdoni/dvote-protobuf/blob/4f28acf85d4607bc70213f9a04fdb1acaac42870/src/vochain/vochain.proto#L261

File: src/architecture/protocol/anonymous-voting/zk-census-proof.md [hunk 7]
```diff
@@ -184,27 +188,27 @@ secret keys to the Vochain, avoiding to need the registration phase.
...
   User->>User: 2. generate CensusRegisterProof of type EthStorageProof
 end
 User->>User: 3. generate zkCensusKey
 User->>Vochain: 4. register zkCensusKey using CensusRegisterProof (registerKeyTx)
 Note over Organizer,User: From here continues with the normal flow
 ```
 
-- 1. *[O+V]* Create **new voting process** (newProcessTx) & define the CensusRegisterFunction=EthStorageProofsCensus
-    - CensusRegisterFunction: checks that:
+- 1. *[O+V]* Create **new voting process** (newProcessTx) & define the CensusOrigin=EthStorageProofsCensus
```
Comment (ed255) [inline src/architecture/protocol/anonymous-voting/zk-census-proof.md:199]:
```suggestion
- 1. *[O+V]* Create **new voting process** (newProcessTx) & define the CensusOrigin=ERC20
```

**Review by ed255 — CHANGES_REQUESTED** [review]

File: src/architecture/protocol/anonymous-voting/zk-census-proof.md (hunk 5 above)
Comment (arnaucube) [reply src/architecture/protocol/anonymous-voting/zk-census-proof.md:167]: No, here it's referring to the *Ethereum Root*, as the *Vochain* will need to check the *MerkleProof* (*EthStorageProof*) with the defined *Ethereum Root*.

**Review by ed255 — APPROVED** [review]

**Pull request merged**